ecosystem/supersim.md

```diff
@@ -0,0 +1,319 @@
+    6. Orchestrator monitors status of running local anvil instances.
+    
+
+### Interfaces
+
+#### Orchestrator interface provides the following to the developer: 
+    
+    1. Instance status
+    2. Event log
+    3. Mode selection
+    4. Control panel (in cli? Some ui option? both?): buttons/commands to start/stop/restart individual instances
+    5. Genesis state interface: could be a long term goal to give devs more control over what they do here. Would offer an interface for applying and managing genesis states: 
+        
+        i. File upload: upload genesis files
+        ii. Genesis history: list of applied genesis states with timestamps/details
+        iii. Apply button/command: apply the selected genesis state to the chosen instances.
+    
+    6. Configuration loader: interface for loading and managing chain configurations from the chain-config-loader. 
+        
+        i. Config list: display available configurations from the registry
+        ii. Download button/command for the selected configuration
+        iii. Apply button/command
+        iv. Configuration details: show detailed information about the selected configuration
+
+#### Op-simulator interface 
+
+Checks instance status as served by orchestrator interface and listens for RPC calls that require op-simulator intervention. The purpose of this interface is to intercept RPC calls that should go to the interface rather than to anvil instances directly. 
+
+This is essentially a proxy between the app's RPC calls and the op-simulator. 
+
+```mermaid
+flowchart TD
+   subgraph Orchestrator Interface
+       direction LR
+       A[Instance Status] -->|Displays| A1[Running Instances]
+       A -->|Displays| A2[Stopped Instances]
+      
+       B[Event Log] -->|Shows| B1[Recent Events]
+       B -->|Shows| B2[System Messages]
+      
+       C[Mode Selection] -->|Options| C1[Vanilla Mode]
+       C -->|Options| C2[Forking Mode]
+      
+       subgraph Control Panel
+           direction TB
+           D1[CLI Commands] -->|Command Line| D1a[start]
+           D1 -->|Command Line| D1b[stop]
+           D1 -->|Command Line| D1c[restart]
+          
+           D2[UI Buttons] -->|Graphical Interface| D2a[start]
+           D2 -->|Graphical Interface| D2b[stop]
+           D2 -->|Graphical Interface| D2c[restart]
+       end
+   end
+```
+
+```mermaid
+flowchart TD
+  subgraph Op-Simulator Interface
+      direction LR
+
+
+      C -->|any other RPC call| D2[call eth_sendRawTransaction \non Anvil]
+
+
+      C[ETH RPC API] -->|eth_sendRawTransaction| C1[Does TX have \n ExecutingMessage?]
+             C1 -->|no| D2
+
+
+      C1 -->|yes| D1
+
+
+    
+          D1[Check Interop Message Invariants] -->|If Pass|D2
+          D1 -->|If Fail| D3[Return TX failed]
+  
+    
+      D2 --> B3[Return result]
+  end
+```
+
+### Services
+
+Given the application flow, the following services will be part of supersim: 
```

Comment:
Will this all still be packaged into the same executable? I don't see a user installing and running 6 different tools

Reply:
Yes 100%, the orchestrator spins these up but it is all "supersim."